tilelang/distributed/utils.py

```
@@ -246,3 +248,19 @@ def CUDA_CHECK(err):
             raise RuntimeError(f"Cuda Error: {err}: {cudart.cudaGetErrorString(err)}")
     else:
         raise RuntimeError(f"Unknown error type: {err}")
+
+
+@lru_cache
+def supports_p2p_native_atomic():
+    """Check if native atomic operations are supported for peer-to-peer (P2P) access between CUDA devices 0 and 1."""
+
+    assert torch.cuda.is_available() and torch.cuda.device_count() > 1
```

Ruff (0.12.2)

Use of `assert` detected

(S101)